src/file.ts

```diff
@@ -1235,9 +1237,9 @@ class File extends ServiceObject<File> {
                   rawResponseStream.pipe(pumpify.obj(throughStreams));
             }
 
-            rawResponseStream.on('end', onComplete).pipe(throughStream, {
-              end: false
-            });
+            rawResponseStream.on('error', onComplete)
+                .on('end', onComplete)
```

Comment:
Are there any scenarios where `onComplete` could fire multiple times? I sort of wonder if we should be using something like `once` to ensure that doesn't happen somehow.

Reply:
I 'spose anything is possible! I added new code for that, please take a look!